utils/ssrf.py

```
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+import socket
+from ipaddress import ip_address
+from typing import Any
+from urllib.parse import urlparse
+
+
+class PinnedResolver:
+    """aiohttp resolver that only returns pre-validated addresses for one host."""
+
+    def __init__(self, addresses_by_host: dict[str, list[str]]):
```

Ruff (0.15.21)

[warning] Missing return type annotation for special method `__init__`

Add return type annotation: `None`

(ANN204)